tests/artemis-test-support/src/main/java/org/apache/activemq/transport/amqp/client/AmqpSender.java

```diff
@@ -71,6 +71,7 @@
 
    private final Set<Delivery> pending = new LinkedHashSet<>();
    private byte[] encodeBuffer = new byte[1024 * 8];
+   DeliveryUpdateListener deliveryUpdateListener;
```

Comment:
I think it might make more sense to extend the validator bits already in the client to add event points for Delivery and Delivery update vs adding yet another callback mechanism.